dist/src/main/java/io/camunda/application/commons/console/ping/PingConsoleTask.java

```diff
@@ -0,0 +1,146 @@
+
+  public static final DateTimeFormatter DATE_TIME_FORMATTER =
+      DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss'Z'").withZone(ZoneOffset.UTC);
+  private static final Logger LOGGER = LoggerFactory.getLogger(PingConsoleTask.class);
+  private static final int NUMBER_OF_MAX_RETRIES = 10;
+  private static final int RETRY_DELAY_MULTIPLIER = 2;
+  private static final Duration MAX_RETRY_DELAY = Duration.ofMinutes(2);
+  public final HttpClient client;
+  private final ManagementServices managementServices;
+  private final ConsolePingConfiguration pingConfiguration;
+  private final RetryDecorator retryDecorator;
+
+  @VisibleForTesting
+  public PingConsoleTask(
+      final ManagementServices managementServices,
+      final ConsolePingConfiguration pingConfiguration,
+      final HttpClient client) {
+    this.managementServices = managementServices;
+    this.pingConfiguration = pingConfiguration;
+    this.client = client;
+    final var retryConfiguration = new RetryConfiguration();
+    retryConfiguration.setMaxRetries(NUMBER_OF_MAX_RETRIES);
+    retryConfiguration.setRetryDelayMultiplier(RETRY_DELAY_MULTIPLIER);
+    retryConfiguration.setMaxRetryDelay(MAX_RETRY_DELAY);
+    retryDecorator = new RetryDecorator();
+  }
+
+  public PingConsoleTask(
+      final ManagementServices managementServices,
+      final ConsolePingConfiguration pingConfiguration) {
+    this(managementServices, pingConfiguration, HttpClient.newHttpClient());
+  }
+
+  @Override
+  public void run() {
+    try {
+      final HttpRequest request =
+          HttpRequest.newBuilder()
+              .uri(URI.create(pingConfiguration.endpoint()))
+              .POST(HttpRequest.BodyPublishers.ofString(getLicensePayload()))
+              .build();
+
+      retryDecorator.decorate("Ping console.", () -> tryPingConsole(request));
+
+    } catch (final Exception e) {
+      if (e instanceof JsonProcessingException) {
+        LOGGER.warn("Failed to parse payload for Console ping task: {}", e.getMessage());
+      } else {
+        LOGGER.warn(
+            "Failed to execute Console ping task. Exception Message: {}", e.getMessage(), e);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  protected void tryPingConsole(final HttpRequest request) throws RetriableException {
+    final HttpResponse<String> resp;
+    try {
+      resp = client.send(request, BodyHandlers.ofString());
+    } catch (final IOException | InterruptedException e) {
+      // If the request fails due to a network issue, we should retry.
+      throw new RetriableException("Network error: " + e.getMessage());
+    }
+
+    // We should retry on server errors, timeouts or too many request.
+    if (resp.statusCode() >= 500) {
+      throw new RetriableException("Server error: " + resp.statusCode(), resp.body());
+    } else if (resp.statusCode() == 429 || resp.statusCode() == 408) {
+      throw new RetriableException(
+          "Too many requests or timeout: " + resp.statusCode(), resp.body());
```

Comment:
🙃 I think both of these should log something but at `debug` level instead of `warn`.